deployments/puppet/manifests/init.pp

```diff
@@ -10,7 +10,6 @@
   $splunk_bundle_dir       = $splunk_otel_collector::params::splunk_bundle_dir,
   $splunk_collectd_dir     = $splunk_otel_collector::params::splunk_collectd_dir,
   $splunk_memory_total_mib = '512',
-  $splunk_ballast_size_mib = '',
```

Comment:
Instead of removing the `splunk_ballast_size_mib` parameter, maybe we should just add a new parameter for configuring `GOMEMLIMIT`, and add some logic to determine which to configure based on the `collector_version` parameter?  We can then just document that `splunk_ballast_size_mib` is only applicable for versions < 0.97.0.

Reply:
GOMEMLimit is acceptable for all version. In the code, we have 90% of the total memory limit assigned if GOMEMLIMIT is not set. We can do the same.

Reply:
But, I have added the logic to have` GOMEMLIMIT` only for 0.97.0 or greater.